pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -214,15 +211,42 @@ def execute_wrapper_jvp(primals, tangents):
         # Backward: branch off based on the gradient function is a device method.
         if isinstance(gradient_fn, qml.gradients.gradient_transform):
             # Gradient function is a gradient transform
+            if _n == max_diff:
 
-            res_from_callback = _grad_transform_jac_via_callback(params, device)
-            if len(tapes) == 1:
-                res_from_callback = [res_from_callback]
+                res_from_callback = _grad_transform_jac_via_callback(params, device)
+                if len(tapes) == 1:
+                    res_from_callback = [res_from_callback]
 
-            jvps = _compute_jvps(res_from_callback, tangents[0], multi_measurements)
+                jvps = _compute_jvps(res_from_callback, tangents[0], multi_measurements)
+            else:
+
+                new_tapes = [_copy_tape(t, a) for t, a in zip(tapes, params)]
+
+                all_jacs = []
+                for new_t in new_tapes:
+                    jvp_tapes, res_processing_fn = gradient_fn(
+                        new_t, shots=device.shots, **gradient_kwargs
+                    )
+                    jacs = execute_tuple(
+                        jvp_tapes,
+                        device,
+                        execute_fn,
+                        gradient_fn,
+                        gradient_kwargs,
+                        _n=_n + 1,
+                        max_diff=max_diff,
+                    )
+                    jacs = res_processing_fn(jacs)
+                    all_jacs.append(jacs)
+
+                if len(all_jacs) == 1 and len(tangents[0]) > 1:
+                    all_jacs = all_jacs[0]
+
+                jvps = _compute_jvps(all_jacs, tangents[0], multi_measurements)
```

Comment:
I think ideally you'd want to return the jacobian itself from the callback, not the `jvp`.

This is because if you only compute the jacobian in the callback, but perform the jacobioan-vector product in "jax" code, jax will trace that product and he will be able to transpose it (it's just a product, so it's linear).

Being able to transpose it means that jax will be able to compute the reverse differentiation (vjp) automatically, simply by swapping the `jacobian-vector` of this function with `vector-jacobian` products.

I believe if you did that, you'd be able to have a single rule for both forward and backward instead of 2 different rules.

In short, I'd suggest to return `all_jacs` from the callback and run `_compute_jvps` in jax-traced code.

Reply:
Hi @PhilipVinc thanks for the comment :slightly_smiling_face: 

If I understand the suggestion correctly, that should already be the case:
* `all_jacs` is the result of the function evaluated using the callback;
* `_compute_jvps` that computs JVPs doesn't perform callbacks anymore;
* both `jax.jacrev` and `jax.jacfwd` should work now.

The reason why there's a "backward" and "forward" mode implementation is specific to the adjoint differentiation method to computing quantum gradients. The adjoint differentiation technique is suited to simultaneously evaluate the quantum function and compute the Jacobian on QNode evaluation. This is not always desired, however, hence PennyLane offers two modes.

Reply:
> all_jacs is the result of the function evaluated using the callback;

Ops sorry, I missed that. Then it looks great!